src/Markdown.js

```diff
@@ -16,20 +16,27 @@ limitations under the License.
 
 import commonmark from 'commonmark';
 import {escape} from "lodash";
+import SdkConfig from './SdkConfig';
 
-const ALLOWED_HTML_TAGS = ['sub', 'sup', 'del', 'u'];
+const ALLOWED_HTML_TAGS = ['sub', 'sup', 'del', 'u', 'code'];
 
 // These types of node are definitely text
 const TEXT_NODES = ['text', 'softbreak', 'linebreak', 'paragraph', 'document'];
 
 function is_allowed_html_tag(node) {
+    if (SdkConfig.get()['latex_maths'] &&
+        node.literal.match(/^<\/?(div|span)( data-mx-maths="[^"]*")?>$/) != null) {
+        return true;
+    }
+
     // Regex won't work for tags with attrs, but we only
     // allow <del> anyway.
     const matches = /^<\/?(.*)>$/.exec(node.literal);
     if (matches && matches.length == 2) {
         const tag = matches[1];
         return ALLOWED_HTML_TAGS.indexOf(tag) > -1;
     }
+
     return false;
 }
 
```

Comment:
As discussed in the room, there is *something* that is causing the parser to fail to catch just a plain `$$$x$$$` is math.  If there is (non-whitespace) text before (e.g. `foo $$$x$$$`), then it works fine.  I suspect that it is something in the `isPlainText` function in this file, possibly around the `if (TEXT_NODES.indexOf(node.type) > -1)` clause that causes it to think that there's no markup, so it just emits the plain text version.

Reply:
Something similar happens when you enter `<div> **foo** </div>` -- it results in the message being sent as text without any markdown processing (the `foo` is not bolded).  That indicates that the bug isn't with your code, but math markup makes it more obvious since nobody is going to type `<div> **foo** </div>`.  So it probably makes sense to fix it as part of this PR (assuming they're caused by the same thing).

Reply:
This is now fixed for the case of latex math in 2204e6c. The case you pointed out actually doesn't see to be a bug, but the expected (but weird) behaviour of HTML blocks, according to the Commonmark spec: https://spec.commonmark.org/0.29/#html-blocks. This was pointed out by @rda in #pigeon.digital:matrix.org.